spec/unit/file_serving/http_metadata_spec.rb

```diff
@@ -29,6 +29,14 @@
       expect( metadata.mode ).to be_nil
     end
 
+    it "skips md5 checksum type in collect on FIPS enabled platforms" do
+      allow(Puppet::Util::Platform).to receive(:fips_enabled?).and_return(true)
+      metadata = described_class.new(http_response)
+      allow(http_response).to receive(:[]).with('X-Checksum-Md5').and_return("c58989e9740a748de4f5054286faf99b")
+      metadata.collect
+      expect( metadata.checksum_type ).to eq :mtime
+    end
+
```

Comment:
If I revert 1f7c8ef64843847846b0a8b2d8a1fe9858eb0bd8 then the test still passes when it shouldn't. I'm thinking something is not quite right about the way the response is stubbed.